yew-ui/src/components/host.rs

```diff
@@ -50,189 +55,206 @@ pub struct MeetingProps {
-                        ))
-                        .unwrap();
-                    let audio_reader = audio_processor
-                        .readable()
-                        .get_reader()
-                        .unchecked_into::<ReadableStreamDefaultReader>();
-                    let mut counter = 0;
-                    let poll_video = async {
-                        loop {
-                            if destroy.load(Ordering::Acquire) {
-                                return;
-                            }
-                            match JsFuture::from(video_reader.read()).await {
-                                Ok(js_frame) => {
-                                    let video_frame =
-                                        Reflect::get(&js_frame, &JsString::from("value"))
-                                            .unwrap()
-                                            .unchecked_into::<VideoFrame>();
-                                    let mut opts = VideoEncoderEncodeOptions::new();
-                                    counter = (counter + 1) % 50;
-                                    opts.key_frame(counter == 0);
-                                    video_encoder.encode(&video_frame);
-                                    video_frame.close();
-                                }
-                                Err(e) => {
-                                    log!("error", e);
-                                }
-                            }
-                        }
-                    };
-                    let poll_audio = async {
-                        loop {
-                            if destroy.load(Ordering::Acquire) {
-                                return;
-                            }
-                            match JsFuture::from(audio_reader.read()).await {
-                                Ok(js_frame) => {
-                                    let audio_frame =
-                                        Reflect::get(&js_frame, &JsString::from("value"))
-                                            .unwrap()
-                                            .unchecked_into::<AudioData>();
-                                    on_audio.emit(audio_frame);
-                                }
-                                Err(e) => {
-                                    log!("error", e);
-                                }
-                            }
-                        }
-                    };
-                    join!(poll_video, poll_audio).await;
-                });
+        match (&self.state, msg) {
+            (&State::Initialized, Msg::Start) => false,
+            (&State::Created, Msg::Start) => {
+                self.initialize(ctx);
                 true
             }
         }
     }
 
     fn view(&self, ctx: &Context<Self>) -> Html {
         // Initialize component on first render.
-        if !self.initialized {
+        if let State::Created = self.state {
```

Comment:
@JasterV We found an easier way to do this using the yew rendered method: 

https://github.com/security-union/rust-zoom/blob/dde8d10d568e1b382cee8f61f749cab98b2414c1/yew-ui/src/components/host.rs#L62-L66

so we removed this state flag all together

Reply:
should I remove the state only on the host or also on the attendants?